Application/Features/OfficeFeature/Queries/GetOfficeListQuery.cs

```diff
@@ -29,6 +31,17 @@ public async Task<GetOfficeListQueryResponse> Handle(GetOfficeListQueryRequest q
             var offices = await _context.Offices
                 .ProjectTo<OfficeDto>(_mapper.ConfigurationProvider)
                 .ToListAsync(cancellationToken);
+            if (query.SearchBy is not null)
+            {
+                offices = offices.Where(x => x.Country.Contains(query.SearchBy)
+                                            || x.City.Contains(query.SearchBy)
+                                            || x.Name.Contains(query.SearchBy)).ToList();             
+            }
+
+            if (query.ListQuantity.HasValue)
+            {
+                offices = offices.Take(Convert.ToInt32(query.ListQuantity)).ToList(); 
```

Comment:
you can omit convert. and specify default value in request, like Top = 10, not nullable int, just int and pass it to take

Reply:
done